src/AggregateFunctions/AggregateFunctionSumMap.h

```diff
@@ -250,35 +264,86 @@ class AggregateFunctionMapBase : public IAggregateFunctionDataHelper<
         }
     }
 
-    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf) const override
+    void serialize(ConstAggregateDataPtr __restrict place, WriteBuffer & buf, std::optional<size_t> version) const override
     {
+        if (!version)
+            version = getDefaultVersion();
+
         const auto & merged_maps = this->data(place).merged_maps;
         size_t size = merged_maps.size();
         writeVarUInt(size, buf);
 
+        std::function<void(size_t, const Array &)> serialize;
+        switch (*version)
+        {
+            case 0:
+            {
+                serialize = [&](size_t col_idx, const Array & values){ values_serializations[col_idx]->serializeBinary(values[col_idx], buf); };
+                break;
+            }
+            case 1:
+            {
+                serialize = [&](size_t col_idx, const Array & values)
+                {
+                    const auto & type = values_types[col_idx];
+                    if (!type->isNullable())
+                        type->promoteNumericType()->getDefaultSerialization()->serializeBinary(values[col_idx], buf);
+                    else
+                        values_serializations[col_idx]->serializeBinary(values[col_idx], buf);
```

Comment:
I think, it's better to create promoted serialization in constructor, because it's quite slow to promote type and create serialization for each row.

Reply:
Also we can do it for `Nullable` too. `removeNullable` -> `promoteNumericType` -> `makeNullable`.